db/version_edit.h

```diff
@@ -374,10 +376,31 @@ class VersionEdit {
     return blob_file_garbages_;
   }
 
+  // Add a WAL (either just created or closed).
+  void AddWal(WalNumber number, const WalMetadata& metadata = WalMetadata()) {
```

Comment:
You could consider taking `WalMetadata` by value here and `std::move`'ing it in the method body (same below).

Reply:
there will be a copy anyway, what's the benefit then?